elide-contrib/elide-test-helpers/src/main/java/com/yahoo/elide/contrib/testhelpers/graphql/GraphQLDSL.java

```diff
@@ -0,0 +1,535 @@
+     * @see <a href="https://graphql.org/learn/queries/#arguments">Arguments</a>
+     * @see <a href="https://graphql.org/learn/schema/#object-types-and-fields">Object Types and Fields</a>
+     */
+    public static Selection field(String name, Arguments arguments, SelectionSet... selectionSet) {
+        return new Field(name, arguments, relayWrap(Arrays.asList(selectionSet)));
+    }
+
+    /**
+     * Creates an attribute(scalar field) selection.
+     *
+     * @param name  The name of the selected attribute that would appear in a GraphQL query.
+     *
+     * @return a field that represents an non-reltionship entity attribute
+     *
+     * @see <a href="https://graphql.org/learn/schema/#scalar-types">Scalar Field</a>
+     */
+    public static Selection field(String name) {
+        return Field.scalarField(name);
+    }
+
+    /**
+     * Creates a complete set of {@link Argument}s that is passed to a object field selection.
+     *
+     * @param arguments  A variable-sized group of {@link Argument}s passed to the object field selection
+     *
+     * @return a complete specification of selection arguments
+     *
+     * @see <a href="https://graphql.org/learn/queries/#arguments">Arguments</a>
+     */
+    public static Arguments arguments(Argument... arguments) {
+        return new Arguments(Arrays.asList(arguments));
+    }
+
+    /**
+     * Creates a {@link Argument single argument} that is passed to a object field selection.
+     *
+     * @param argument  The single {@link Argument} passed to the object field selection
+     *
+     * @return a complete specification of selection argument
+     *
+     * @see <a href="https://graphql.org/learn/queries/#arguments">Arguments</a>
+     */
+    public static Arguments argument(Argument argument) {
+        return new Arguments(Collections.singletonList(argument));
+    }
+
+    /**
+     * Creates a spec object of an GraphQL {@link Argument argument} whose value is not quoted.
+     *
+     * @param name  The name of the argument
+     * @param value  An object that models argument value.
+     *
+     * @return a complete specification of selection arguments
+     *
+     * @see <a href="https://graphql.org/learn/queries/#arguments">Arguments</a>
+     */
+    public static Argument argument(String name, Object value) {
+        return argument(name, value, false);
+    }
+
+    /**
+     * Creates a spec object of an GraphQL {@link Argument argument}.
+     *
+     * @param name  The name of the argument
+     * @param value  An object that models argument value.
+     * @param quoted  Whether or not the serialized {@code value} should be double-quote
+     *
+     * @return a complete specification of selection arguments
+     *
+     * @see <a href="https://graphql.org/learn/queries/#arguments">Arguments</a>
+     */
+    public static Argument argument(String name, Object value, boolean quoted) {
+        if (value instanceof String) {
```

Comment:
Whether the argument is a string or not, it should be convertable to JSON - why not just always run it through the JSON parser?